src/sim/mod.rs

```diff
@@ -356,7 +357,7 @@ impl<T: DataSource> BacktestBroker for SimulatedBroker<T> {
                         dividend.value, dividend.symbol
                     );
                     let cash_value = CashValue::from(*qty.clone() * *dividend.value);
-                    self.credit(&cash_value);
+                    self.clone().credit(&cash_value);
```

Comment:
I think this is why the dividends test case is failing (and giving such a weird fail message).

I can't remember what the test looks like but I think this would have to be:

`self.credit(&cash_value);` 

I think you are cloning the broker and then crediting here.

Reply:
Ahhh makes sense

Reply:
Gonna be tricky resolving this issue due to the double borrow that happens, once in the if let and once in the for loop.

Reply:
Way to get around this is to create local variable outside the loop.

```
fn pay_dividends(&mut self) {
        info!("BROKER: Checking dividends");
        let mut dividend_value:CashValue = CashValue::from(0.0);
        if let Some(dividends) = self.data.get_dividends() {
            for dividend in dividends.iter() {
                //Our dataset can include dividends for stocks we don't own so we need to check
                //that we own the stock, not performant but can be changed later
                if let Some(qty) = self.get_position_qty(&dividend.symbol) {
                    info!(
                        "BROKER: Found dividend of {:?} for portfolio holding {:?}",
                        dividend.value, dividend.symbol
                    );
                    let cash_value = CashValue::from(*qty.clone() * *dividend.value);
                    dividend_value = cash_value.clone() + dividend_value;
                    let dividend_paid = DividendPayment::new(
                        cash_value,
                        dividend.symbol.clone(),
                        dividend.date,
                    );
                    self.log.record(dividend_paid);
                }
            }
        }
        self.credit(&dividend_value);
```